From 2285c8b1a96396d3d1a35e07dccf242a2be4c796 Mon Sep 17 00:00:00 2001
From: jianjunz <jianjun.zhu@intel.com>
Date: Wed, 13 Jul 2022 14:48:29 +0800
Subject: [PATCH 3/3] Create transceivers explicitly. (#287)

Transceivers created implicitly by addTrack are not stopped correctly,
which result resource leak.
---
 .../src/main/java/owt/base/PeerConnectionChannel.java     | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/sdk/base/src/main/java/owt/base/PeerConnectionChannel.java b/src/sdk/base/src/main/java/owt/base/PeerConnectionChannel.java
index b1c2094..4444547 100644
--- a/src/sdk/base/src/main/java/owt/base/PeerConnectionChannel.java
+++ b/src/sdk/base/src/main/java/owt/base/PeerConnectionChannel.java
@@ -202,12 +202,12 @@ public abstract class PeerConnectionChannel
             ArrayList<String> streamIds = new ArrayList<>();
             streamIds.add(mediaStream.getId());
             for (AudioTrack audioTrack : mediaStream.audioTracks) {
-                RtpSender audioSender = peerConnection.addTrack(audioTrack, streamIds);
-                audioRtpSenders.put(mediaStream.getId(), audioSender);
+                RtpTransceiver transceiver = peerConnection.addTransceiver(audioTrack, new RtpTransceiver.RtpTransceiverInit(RtpTransceiver.RtpTransceiverDirection.SEND_ONLY, streamIds, new ArrayList<RtpParameters.Encoding>()));
+                audioRtpSenders.put(mediaStream.getId(), transceiver.getSender());
             }
             for (VideoTrack videoTrack : mediaStream.videoTracks) {
-                RtpSender videoSender = peerConnection.addTrack(videoTrack, streamIds);
-                videoRtpSenders.put(mediaStream.getId(), videoSender);
+                RtpTransceiver transceiver = peerConnection.addTransceiver(videoTrack, new RtpTransceiver.RtpTransceiverInit(RtpTransceiver.RtpTransceiverDirection.SEND_ONLY, streamIds, new ArrayList<RtpParameters.Encoding>()));
+                videoRtpSenders.put(mediaStream.getId(), transceiver.getSender());
             }
         });
     }
-- 
2.37.2

